### PR TITLE
Reset Incremental Export checkpoints

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://trello.com/c/DSVBVUPn/12-data-sf-outbound-scheduled-ucr-export

##### SUMMARY

Adds the ability to resend all cases, or all cases after a particular checkpoint for each incremental export.

I got bogged down so much by crispy forms that I gave up and rewrote the UI using things that didn't make me feel so crusty, so this ended up being much bigger than it needed to be. It seemed unreasonably complicated to add a button with an unrelated URL in crispy forms, but maybe I just missed the magic way of doing this. 
The new UI follows more of our conventions, and doesn't require you to click "Delete" when updating a particular item. 

This also fixes a different bug where everyone could see and use all the connection settings for every domain on HQ (!). 

Screenshot from 2020-05-11 14-04-18
Screenshot from 2020-05-11 14-04-45

:blowfish: / :blowfish: 

##### FEATURE FLAG

Incremental Exports

### CODE REVIEW

File: corehq/apps/export/templates/export/incremental_export.html [hunk 1]
```diff
@@ -107,5 +112,38 @@ <h3>
...
+                {% trans "Resend all cases for " %} <strong data-bind="text: name"></strong>?
+              </h3>
+            </div>
+            <div class="modal-footer">
+              <button type="button"
+                      class="btn btn-default"
+                      data-dismiss="modal">
+                {% trans "Cancel" %}
+              </button>
+              <a data-bind="attr: {href: $parents[1].reverse('incremental_export_resend_all', id)}"
```
Comment (orangejenny) [inline corehq/apps/export/templates/export/incremental_export.html:139]: What does `$parents[1]` mean? Would `$root` be equivalent here?

**Review by orangejenny — APPROVED** [review]
This is beautiful. Just minor comments.

File: corehq/apps/export/views/incremental.py [hunk 2]
```diff
@@ -23,55 +22,123 @@
...
+            "template": "base-incremental-export-template",
+        }
 
 
 class IncrementalExportLogView(GenericTabularReport, DatespanMixin):
-    name = 'Incremental Export Logs'
-    base_template = 'reports/base_template.html'
-    section_name = 'Project Settings'
-    slug = 'incremental_export_logs'
+    name = "Incremental Export Logs"
```
Comment (kaapstorm) [inline corehq/apps/export/views/incremental.py:130]: (Curious: You have feelings about quotation marks? Choosing double over single is one of the two more contentious issues about Black (the other being its 88-character line length). Do you have feelings about auto formatters? Do you use one?)

**Review by kaapstorm — APPROVED** [review]
Very nice!

(About that nasty `ConnectionSettings` bug: It was fixed [here](https://github.com/dimagi/commcare-hq/pull/27258). Sorry I didn't ping you on the PR at the time.)

File: corehq/apps/export/views/incremental.py (hunk 2 above)
Comment (proteusvacuum) [reply corehq/apps/export/views/incremental.py:130]: I don't. I actually ran black on that file, but probably shouldn't.

**Comment by proteusvacuum** [comment]
@kaapstorm, I think that PR only fixes the bug for when you create connection settings. In this PR, it also fixes it for the Incremental Export dropdown. If you look on prod right now, you can still see all connection settings for every domain on that page.

File: corehq/apps/export/templates/export/incremental_export.html (hunk 1 above)
Comment (proteusvacuum) [reply corehq/apps/export/templates/export/incremental_export.html]: yes, thanks!

File: corehq/apps/export/forms.py [hunk 3]
```diff
@@ -1096,60 +1091,73 @@ class Meta:
...
+            crispy.Field('export_instance_id'),
+            crispy.Field('connection_settings'),
+            crispy.Field('active'),
+            hqcrispy.FormActions(
+                twbscrispy.StrictButton(
+                    ugettext_lazy("Update"),
+                    css_class='btn-primary',
+                    type='submit',
+                ),
+                crispy.HTML('<button type="button" class="btn" data-dismiss="modal">Cancel</button>'),
```
[the remark this answers is not in the document]
Comment (proteusvacuum) [reply corehq/apps/export/forms.py]: Also added this to the docs

**Comment by kaapstorm** [comment]
Wow. Thank you.

**Pull request merged**